File: app/src/main/java/com/example/doan/Models/UserProfileDto.java
package com.example.doan.Models;

import com.google.gson.annotations.SerializedName;

public class UserProfileDto {

    @SerializedName("id")
    private Long id;

    @SerializedName("username")
    private String username;

    @SerializedName("fullName")
    private String fullName;

    @SerializedName("email")
    private String email;

    @SerializedName("phone")
    private String phone;

    @SerializedName("address")
    private String address;

    @SerializedName("memberTier")
    private String memberTier;

    @SerializedName("points")
    private Integer points;

    // Getters
    public Long getId() {
        return id;
    }

    public String getUsername() {
        return username;
    }

    public String getFullName() {
        return fullName;
    }

    public String getEmail() {
        return email;
    }

    public String getPhone() {
        return phone;
    }

    public String getAddress() {
        return address;
    }

    public String getMemberTier() {
        return memberTier;
    }

    public Integer getPoints() {
        return points;
    }
}

File: app/src/main/java/com/example/doan/Models/UserProfileDto.kt
package com.example.doan.Models

import com.google.gson.annotations.SerializedName

data class UserProfileDto(
    @SerializedName("id")
    val id: Long? = null,

    @SerializedName("username")
    val username: String? = null,

    @SerializedName("fullName")
    val fullName: String? = null,

    @SerializedName("email")
    val email: String? = null,

    @SerializedName("phone")
    val phone: String? = null,

    @SerializedName("address")
    val address: String? = null,

    @SerializedName("memberTier")
    val memberTier: String? = null,

    @SerializedName("points")
    val points: Int? = null,

    @SerializedName("avatarUrl")
    val avatar: String? = null,

    @SerializedName("role")
    val role: String? = null
)
